imap_processing/tests/ultra/unit/test_ultra_l1b_extended.py

```diff
@@ -0,0 +1,243 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from imap_processing import decom
+from imap_processing.cdf.defaults import GlobalConstants
+from imap_processing.ultra.l0.decom_ultra import process_ultra_apids
+from imap_processing.ultra.l0.ultra_utils import (
+    ULTRA_AUX,
+    ULTRA_EVENTS,
+)
+from imap_processing.ultra.l1a.ultra_l1a import create_dataset
+from imap_processing.ultra.l1b.ultra_l1b_extended import (
+    determine_species_pulse_height,
+    determine_species_ssd,
+    get_coincidence_positions,
+    get_energy_ssd,
+    get_front_x_position,
+    get_front_y_position,
+    get_particle_velocity,
+    get_path_length,
+    get_ph_tof_and_back_positions,
+    get_ssd_offset_and_positions,
+    get_ssd_tof,
+)
+from imap_processing.utils import group_by_apid
+
+
+@pytest.fixture()
+def de_dataset(ccsds_path_theta_0, xtce_path):
```

Comment:
Is this fixture the same as in test_ultra_l1b.py? If so, it can be moved into a conftest.py and scope set to "sesson" so that the output is cached and not regenerated each test it is used in.